### DISCUSSION TITLE
Render a notion page

### DISCUSSION THREAD

**Discussion by jeff1985** [mariosimao/notion-sdk-php#385] [Q&A] [answered]
Hi! 

Is there an easy way of rendering a notion page to html?

I am able to load the page like this:

    $pageId = "xx";
    $page = $notion->pages()->find($pageId);
    echo $page->title()?->toString();
    $blocks = $notion->blocks()->findChildren($pageId);

But is there any way to automatically convert it to html?

**Comment by mariosimao** [comment, marked as answer]
Hi @jeff1985! 

The notion-sdk-php has a Markdown renderer.

```php
<?php

require "vendor/autoload.php";

use Notion\Blocks\Renderer\MarkdownRenderer;
use Notion\Notion;

$token = "secret_token";
$notion = Notion::create($token);

$pageId = "page_id";
$blocks = $notion->blocks()->findChildrenRecursive($pageId);

echo PHP_EOL;
echo MarkdownRenderer::render(...$blocks);
```

The feature could mixed with a Markdown to HTML converter like [CommonMark](https://commonmark.thephpleague.com).

For more info, see #197
